src/main/java/mate/academy/security/CustomUserDetailsService.java

```diff
@@ -0,0 +1,21 @@
+package mate.academy.security;
+
+import lombok.RequiredArgsConstructor;
+import mate.academy.repository.user.UserRepository;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class CustomUserDetailsService implements UserDetailsService {
+    private final UserRepository userRepository;
+
+    @Override
+    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
+        return userRepository.findByEmail(email)
+                .orElseThrow(
+                        () -> new UsernameNotFoundException("User not found"));
```

Comment:
```suggestion
                        () -> new EntityNotFoundException("User not found"));
```

add email to message